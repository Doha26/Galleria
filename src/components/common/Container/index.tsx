import React from 'react';
import {ScrollView, SafeAreaView, View} from 'react-native';
import PropTypes from 'prop-types';
import colors from '~/theming/colors';

const propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.arrayOf(PropTypes.node)]).isRequired,
  onScroll: PropTypes.func,
  transparency: PropTypes.bool,
  withSafeArea: PropTypes.bool,
  withScroll: PropTypes.bool,
};

const defaultProps = {};

const renderContent = (
  withSafeArea: boolean | undefined,
  children: React.ReactNode,
  onScroll: ((offset: number) => null) | undefined,
  withScroll: boolean | undefined,
) => {
  if (withScroll) {
    if (withSafeArea) {
      return (
        <SafeAreaView style={{flex: 1, backgroundColor: colors.transparent}}>
          <ScrollView
            scrollEventThrottle={1}
            onScroll={({nativeEvent}) => (onScroll ? onScroll(nativeEvent.contentOffset.y) : null)}
            style={{flex: 1}}>
            {children}
          </ScrollView>
        </SafeAreaView>
      );
    }
    return (
      <ScrollView
        scrollEventThrottle={1}
        onScroll={({nativeEvent}) => (onScroll ? onScroll(nativeEvent.contentOffset.y) : null)}
        style={{flex: 1}}>
        {children}
      </ScrollView>
    );
  }
  return <View style={{flex: 1}}>{children}</View>;
};
const Container = ({
  onScroll,
  children,
  withSafeArea,
  withScroll,
}: {
  onScroll?: (offset: number) => null;
  children?: React.ReactNode;
  transparency?: boolean;
  withSafeArea?: boolean;
  withScroll?: boolean;
}) => {
  return renderContent(withSafeArea, children, onScroll, withScroll);
};

Container.propTypes = propTypes;
Container.defaultProps = defaultProps;

export default Container;
