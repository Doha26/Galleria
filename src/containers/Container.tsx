import * as React from 'react';
import PropTypes from 'prop-types';
import {StyleSheet, View} from 'react-native';
import {padding} from '~/components/config/spacing';
import ViewPropTypes from '../components/config/ViewPropTypes';

const Container = ({
  style,
  children,
  ...rest
}: {
  disable: boolean;
  style: object;
  children: any;
}) => {
  return (
    <View style={StyleSheet.flatten([styles.container, style])} {...rest}>
      {children}
    </View>
  );
};

const styles = {
  container: {
    paddingHorizontal: padding.large,
  },
  right: {
    paddingRight: 0,
  },
  left: {
    paddingLeft: 0,
  },
  all: {
    paddingHorizontal: 0,
  },
};

Container.propTypes = {
  disable: PropTypes.bool,
  style: ViewPropTypes.style,
  children: PropTypes.node,
};
Container.defaultProps = {
  disable: false,
  style: {},
  children: null,
};

export default Container;
