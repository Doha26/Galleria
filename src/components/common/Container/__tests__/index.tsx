import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Container from '~/components/common/Container';

describe('Container Component', () => {
  it('should render without issues', () => {
    const component = shallow(<Container withScroll />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
