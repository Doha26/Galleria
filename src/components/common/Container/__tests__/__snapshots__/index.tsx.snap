// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component should render without issues 1`] = `
<Component
  onScroll={[Function]}
  scrollEventThrottle={1}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
